frontend/src/components/ExperimentsTableConfigurator.jsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  FormGroup,
+  Label,
+  Input,
+  Form
+} from 'reactstrap';
+
+class ExperimentsTableConfigurator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleModalShow = this.handleModalShow.bind(this);
+    this.handleModalHide = this.handleModalHide.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+
+    this.state = {
+      showModal: false
+    };
+  }
+
+  handleModalShow() {
+    this.setState({
+      showModal: true
+    });
+  }
+
+  handleModalHide() {
+    this.setState({
+      showModal: false
+    });
+  }
+
+  handleChange(prefix, event) {
+    const { projectConfig } = this.props;
+    const { tableState } = projectConfig;
+    const {
+      hiddenLogKeys = [],
+      hiddenArgKeys = []
+    } = tableState;
+
+    if (prefix === 'logKey') {
+      const nextHiddenLogKeys = !event.target.checked
+        ? hiddenLogKeys.concat(event.target.name)
+        : hiddenLogKeys.filter((vk) => vk !== event.target.name);
+
+      this.props.onTableColumnsVisibilityUpdate(
+        this.props.project.id, nextHiddenLogKeys, hiddenArgKeys);
+    } else {
+      const nextHiddenArgKeys = !event.target.checked
+        ? hiddenArgKeys.concat(event.target.name)
+        : hiddenArgKeys.filter((vk) => vk !== event.target.name);
+
+      this.props.onTableColumnsVisibilityUpdate(
+        this.props.project.id, hiddenLogKeys, nextHiddenArgKeys);
+    }
+  }
+
+  render() {
+    const { stats = {}, projectConfig } = this.props;
+    const { tableState } = projectConfig;
+    const {
+      logKeys = [],
+      argKeys = []
+    } = stats;
+    const {
+      hiddenLogKeys = [],
+      hiddenArgKeys = []
+    } = tableState;
+
+    return (
+      <div>
+        <Button color="secondary" className="my-2" onClick={this.handleModalShow}>
+          Table settings
+        </Button>
+
+        <Modal isOpen={this.state.showModal}>
+          <ModalHeader>
+            Edit table columns visibility
+          </ModalHeader>
+          <ModalBody>
+            <Form>
+              <legend>Log Keys</legend>
+              {
+                logKeys.map((l) => (
+                  <FormGroup check key={`logKey.${l}`}>
+                    <Label check>
+                      <Input
+                        type="checkbox"
+                        name={l}
+                        checked={!hiddenLogKeys.some((vk) => vk === l)}
```

Comment:
I think hiddenLogKeys does not include falsy value such as empty string.
I think it's easier to use `find` method than `some` method.